feature/s3/transfermanager/api_op_GetObject.go

```diff
@@ -0,0 +1,940 @@
+// files into smaller parts/ranges according to config and getting them in sequence.
+// You can configure the download type and chunk size through the Options parameters.
+//
+// Additional functional options can be provided to configure the individual
+// download. These options are copies of the original Options instance, the client of which GetObject is called from.
+// Modifying the options will not impact the original Client and Options instance.
+func (c *Client) GetObject(ctx context.Context, input *GetObjectInput, opts ...func(*Options)) (*GetObjectOutput, error) {
+	i := downloader{in: input, options: c.options.Copy()}
+	for _, opt := range opts {
+		opt(&i.options)
+	}
+	i.options.Concurrency = 1
+
+	return i.download(ctx)
+}
+
+// DownloadObject downloads an object from S3, intelligently splitting large
+// files into smaller parts/ranges according to config and getting them in parallel across
+// multiple goroutines. You can configure the download type, chunk size and concurrency
+// through the Options parameters.
+//
+// Additional functional options can be provided to configure the individual
+// download. These options are copies of the original Options instance, the client of which DownloadObject is called from.
+// Modifying the options will not impact the original Client and Options instance.
+func (c *Client) DownloadObject(ctx context.Context, w io.WriterAt, input *GetObjectInput, opts ...func(*Options)) (*GetObjectOutput, error) {
+	i := downloader{in: input, options: c.options.Copy(), w: w}
+	for _, opt := range opts {
+		opt(&i.options)
+	}
+
+	return i.download(ctx)
+}
+
+type downloader struct {
+	options Options
+	in      *GetObjectInput
+	out     *GetObjectOutput
+	w       io.WriterAt
+	buf     []byte
+
+	wg sync.WaitGroup
+	m  sync.Mutex
+
+	offset     int64
+	pos        int64
+	totalBytes int64
+	written    int64
+
+	err error
+}
+
+func (d *downloader) download(ctx context.Context) (*GetObjectOutput, error) {
+	if err := d.init(ctx); err != nil {
+		return nil, fmt.Errorf("unable to initialize download: %w", err)
+	}
+
+	clientOptions := []func(*s3.Options){
+		func(o *s3.Options) {
+			o.APIOptions = append(o.APIOptions,
+				middleware.AddSDKAgentKey(middleware.FeatureMetadata, userAgentKey),
+				addFeatureUserAgent,
+			)
+		}}
+
+	if d.in.PartNumber > 0 {
+		return d.singleDownload(ctx, clientOptions...)
+	}
+
+	var output *GetObjectOutput
+	if d.options.MultipartDownloadType == types.MultipartDownloadTypePart {
+		if d.in.Range != "" {
+			return d.singleDownload(ctx, clientOptions...)
+		}
+		output = d.getChunk(ctx, 1, "", clientOptions...)
+		if d.getErr() != nil {
+			return output, d.err
+		}
+
+		if output.PartsCount > 1 {
+			partSize := output.ContentLength
+			ch := make(chan dlchunk, d.options.Concurrency)
+			for i := 0; i < d.options.Concurrency; i++ {
+				d.wg.Add(1)
+				go d.downloadPart(ctx, ch, clientOptions...)
+			}
+
+			for i := int32(2); i <= output.PartsCount; i++ {
```

Comment:
why do we start at 2 here?

Reply:
part 1 should already be written to output when we call `getChunk()` to get total parts above